File: hospital-Management-System-Backend/hospital-Management-System-Backend/src/main/java/com/project/hospital_Management_System_Backend/dtos/EmployeeRequestDto.java
package com.project.hospital_Management_System_Backend.dtos;

import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Data
public class EmployeeRequestDto {
    private CredentialsDto credentials;

    private ProfileDto profile;

    private boolean admin;

    private boolean doctor;
}

File: frontend/hospital-Management-System-frontend/src/app/models/EmployeeRequestDTO.ts
import CredentialDTO from "./CredentialDTO";
import ProfileDTO from "./ProfileDTO";

export default interface EmployeeRequestDTO{
    credentials:CredentialDTO,
    profile:ProfileDTO,
    age:number,
    gender:string,
    address:string,
    admin:boolean,
    doctor:boolean,
    specialty:string,
    salary:number
}